Src/Support/Google.Apis.Auth/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -204,21 +204,13 @@ internal GoogleCredential CreateDefaultCredentialFromJson(string json)
 
 
         /// <summary>Creates a default credential from JSON data.</summary>
-        private static GoogleCredential CreateDefaultCredentialFromParameters(JsonCredentialParameters credentialParameters)
-        {
-            switch (credentialParameters.Type)
+        internal GoogleCredential CreateDefaultCredentialFromParameters(JsonCredentialParameters credentialParameters) =>
```

Comment:
Does it need to be an instance method, or is to keep aligned with the rest of the design (which has always felt a little unusual to me).

Reply:
Just to keep in line with the rest.
When we call this method it could potentially be on a DefaultCredentialProvider derived class.